src/common/components/clusterConfiguration/CertificateFields.tsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import { useField, useFormikContext } from 'formik';
+import { Grid } from '@patternfly/react-core';
+import { CheckboxField, HelperText, PopoverIcon, UploadField } from '../ui';
+import { TrustedCertificateFieldsType } from '../../types';
+import { useTranslation } from '../../hooks/use-translation-wrapper';
+
+import { InfraEnv } from '../../api';
+
+const FIELD_NAME = 'certificateField';
+
+export const CertificateFieldsHelperText = ({ fieldId = FIELD_NAME }) => {
+  const { t } = useTranslation();
+  return (
+    <HelperText fieldId={fieldId}>
+      {t(
+        'ai:Paste one or more PEM formatted certificates that you want the cluster to trust into this field.',
+      )}
+    </HelperText>
+  );
+};
+
+export const CertificateInputFields = () => {
+  const [{ name, value }, , { setValue }] =
+    useField<InfraEnv['additionalTrustBundle']>('trustBundle');
+
+  const { t } = useTranslation();
+  return (
+    <Grid hasGutter>
+      <UploadField
+        label={
+          <>
+            {t('ai:Additional certificates')}{' '}
+            <PopoverIcon
+              bodyContent={t(
+                'ai:You can upload additional trusted certificates if the cluster hosts are in a network with a re-encrypting (MITM) proxy or the cluster needs to trust certificates for other purposes (for example, container image registries)',
+              )}
+            />
+          </>
+        }
+        name={name}
+        helperText={<CertificateFieldsHelperText />}
+        idPostfix="discovery"
+        onBlur={() => value && setValue(value)}
+        dropzoneProps={{
+          accept: '.pem',
+          maxSize: 2048,
+          onDropRejected:
+            ({ setError }) =>
+            () =>
+              setError(t('ai:File not supported.')),
+        }}
+      />
+    </Grid>
+  );
+};
+
+const CertificateFields = () => {
+  const { setFieldValue, values, initialValues } = useFormikContext<TrustedCertificateFieldsType>();
+  const resetCertificate = (isNewlyChecked: boolean) => {
+    if (isNewlyChecked) {
+      setFieldValue(FIELD_NAME, initialValues.trustBundle);
+    } else {
+      setFieldValue(FIELD_NAME, '');
+    }
+  };
+  const { t } = useTranslation();
+  return (
+    <>
+      <CheckboxField
```

Comment:
Unchecking this field should clear the certificates, imo. That's what happens with the Proxy fields, we should keep it consistent.

[certificates.webm](https://user-images.githubusercontent.com/87187179/208108582-14ac2f29-9c70-4965-969d-798e5fb7a34f.webm)

Reply:
Done!